tm2/pkg/bft/consensus/types/round_state.go

```diff
@@ -95,21 +97,32 @@ type HRS struct {
 	Step   RoundStepType `json:"step"`
 }
 
-func (hrs HRS) Compare(other HRS) int {
+func (hrs HRS) Compare(other HRS) ordering.Ordering {
 	if hrs.Height < other.Height {
-		return -1
-	} else if hrs.Height == other.Height {
-		if hrs.Round < other.Round {
-			return -1
-		} else if hrs.Round == other.Round {
-			if hrs.Step < other.Step {
-				return -1
-			} else if hrs.Step == other.Step {
-				return 0
-			}
-		}
+		return ordering.Less
+	}
+
+	if hrs.Height > other.Height {
+		return ordering.Greater
+	}
+
+	if hrs.Round < other.Round {
+		return ordering.Less
+	}
+
+	if hrs.Round > other.Round {
+		return ordering.Greater
+	}
+
+	if hrs.Step < other.Step {
+		return ordering.Less
 	}
-	return 1
+
+	if hrs.Step > other.Step {
+		return ordering.Greater
+	}
+
+	return ordering.Equal
 }
```

Comment:
Thank you! :pray: 

Wdyt about the following:

```go
	// Use first occasion where, in order, any of Height, Round or Step are different
	// for comparison.
	for _, cmp := range [...]int{
		cmp.Compare(hrs.Height, other.Height),
		cmp.Compare(hrs.Round, other.Round),
		cmp.Compare(hrs.Step, other.Step),
	} {
		if cmp != 0 {
			return cmp
		}
	}
	return 0
```

Loopless alternative:

```go
	if n := cmp.Compare(hrs.Height, other.Height); n != 0 {
		return n
	}
	// ...
	return 0
```

Reply:
Either is okay. I don't mind either way.

Reply:
This function is actually dead code.